pkg/cmd/repo/create/http_test.go

```diff
@@ -1,37 +1,345 @@
 package create
 
 import (
+	"net/http"
 	"testing"
 
-	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
 	"github.com/cli/cli/pkg/httpmock"
 	"github.com/stretchr/testify/assert"
 )
 
-func Test_RepoCreate(t *testing.T) {
-	reg := &httpmock.Registry{}
-	httpClient := api.NewHTTPClient(api.ReplaceTripper(reg))
-
-	reg.Register(
-		httpmock.GraphQL(`mutation RepositoryCreate\b`),
-		httpmock.GraphQLMutation(`{}`,
-			func(inputs map[string]interface{}) {
-				assert.Equal(t, inputs["description"], "roasted chestnuts")
-				assert.Equal(t, inputs["homepageUrl"], "http://example.com")
-			}),
-	)
-
-	input := repoCreateInput{
-		Description: "roasted chestnuts",
-		HomepageURL: "http://example.com",
+func Test_repoCreate(t *testing.T) {
```

Comment:
These tests are great, just for completeness I think one case that is missing is creating a repo with a hostname that is not `github.com`.